src/ert/_c_wrappers/enkf/ert_config.py

```diff
@@ -217,21 +227,27 @@ def make_suggestion_list(cls, config_file):
         ).suggest_migrations(config_file)
 
     @classmethod
-    def read_site_config(cls):
-        site_config_parser = ConfigParser()
-        init_site_config_parser(site_config_parser)
-        site_config_content = site_config_parser.parse(site_config_location())
-        return config_content_as_dict(site_config_content, {})
+    def read_site_config(cls, new_parser=USE_NEW_PARSER_BY_DEFAULT):
+        if new_parser:
+            return lark_parse(site_config_location())
+        else:
+            site_config_parser = ConfigParser()
+            init_site_config_parser(site_config_parser)
+            site_config_content = site_config_parser.parse(site_config_location())
+            return config_content_as_dict(site_config_content, {})
 
     @classmethod
-    def read_user_config(cls, user_config_file):
-        site_config_dict = ErtConfig.read_site_config()
-        user_config_parser = ErtConfig._create_user_config_parser()
-        user_config_content = user_config_parser.parse(
-            user_config_file,
-            pre_defined_kw_map=ErtConfig._create_pre_defines(user_config_file),
-        )
-        return config_content_as_dict(user_config_content, site_config_dict)
+    def read_user_config(cls, user_config_file, new_parser=USE_NEW_PARSER_BY_DEFAULT):
```

Comment:
please consider renaming `new_parser` to `use_new_parser`
```suggestion
    def read_user_config(cls, user_config_file, use_new_parser=USE_NEW_PARSER_BY_DEFAULT):
```